pkg/steps/multi_stage.go

```diff
@@ -553,7 +556,10 @@ func (s *multiStageTestStep) runPod(ctx context.Context, pod *coreapi.Pod, notif
 	if _, err := createOrRestartPod(s.podClient.Pods(s.jobSpec.Namespace()), pod); err != nil {
 		return fmt.Errorf("failed to create or restart %q pod: %w", pod.Name, err)
 	}
-	_, err := waitForPodCompletion(ctx, s.podClient.Pods(s.jobSpec.Namespace()), s.eventClient.Events(s.jobSpec.Namespace()), pod.Name, notifier, false)
+	newPod, err := waitForPodCompletion(ctx, s.podClient.Pods(s.jobSpec.Namespace()), s.eventClient.Events(s.jobSpec.Namespace()), pod.Name, notifier, false)
+	if newPod != nil {
+		pod = newPod
+	}
 	s.subTests = append(s.subTests, notifier.SubTests(fmt.Sprintf("%s - %s ", s.Description(), pod.Name))...)
 	if err != nil {
 		linksText := strings.Builder{}
```

Comment:
```suggestion
		linksText := strings.Builder{}
		
		if pod == nil {
			return fmt.Errorf("waitForPodCompletion didn't return a pod")
		}
```

Reply:
Turns out we didn't need this after https://github.com/openshift/ci-tools/pull/1257#discussion_r502311427